### PR TITLE
Enable administrators to split facility matches into new facilities

### PR DESCRIPTION
**Pull request by kellyi**
## Overview

This PR enables administrators to split facility matches into new facilities via a new `/dashboard/splitfacilities` route. The new page has the same facility card as is displayed on the merge and delete UIs, but also displays a "Matches to Split" card on the righthand side which displays details about each match along with a "Split" button used to open a confirmation dialog before creating a new facility. On successful splits, we update the UI to display the new OAR ID and also to filter the match out of the source facility's "Contributors" list.

Connects #593 

## Demo

Screen Shot 2019-06-25 at 7 33 40 PM

## Notes

~Although this might be an edge case I noticed a bug in what I think should be a valid operation: after splitting a match from Facility A into Facility B, I attempted to merge Facility A into Facility B and encountered this error:~

```
("Cannot delete some instances of model 'Facility' because they are referenced through a protected foreign key: 'FacilityListItem.facility'", <QuerySet [<FacilityListItem: FacilityListItem 594 - MATCHED>, <FacilityListItem: FacilityListItem 277 - MATCHED>]>)
```

~Merging B back into A worked as expected, so I'm not quite sure what causes the error above. Any ideas?~

I believe I fixed this and that the problem was that I had neglected to update the facility_list_item.facility to the new facility.
## Testing Instructions

- serve this branch then `resetdb` & `processfixtures`
- sign in as c1 then search for facility name "Regina" in country "China" and copy the OAR ID
- visit the dashboard page, then click on the "Split Facility Matches" link
- search for the OAR you copied
- split the matches and verify that they create a new facility
- you can also try out merging the new facilities back into the original facility, or merging facilities then splitting them then merging them etc

## Checklist

- [ ] `fixup!` commits have been squashed
- [ ] CI passes after rebase
- [ ] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines

### CODE REVIEW

**Comment by jwalgran** [comment]
Looking at this now.

File: src/django/api/views.py [hunk 1]
```diff
@@ -941,38 +941,81 @@ def merge(self, request):
...
+            match_for_new_facility.status = FacilityMatch.CONFIRMED
+            match_for_new_facility.results = {
+                'match_type': 'split_by_administator',
+                'split_from_oar_id': match_for_new_facility.facility.id,
+            }
+
+            match_for_new_facility.save()
+
+            list_item_for_match.facility = new_facility
+            list_item_for_match.save()
```
Comment (jwalgran) [inline src/django/api/views.py]: The delete and merge APIs add an item to `processing_results`, which acts as an audit trail of all the mutations that have been applied to the line item. Consider continuing that pattern here following the example of merge:

https://github.com/open-apparel-registry/open-apparel-registry/blob/95395af3c6d2a30dedc350902fe27ac5453e82cc/src/django/api/views.py#L909-L915

We would want to add a new `ProcessingAction.SPLIT_MATCH` constant.

Comment (kellyi) [reply src/django/api/views.py]: Sounds good! I added this in fc74dbe and the entry now shows "split_match" as the action and also indicates the "previous_facility_oar_id":

```
>>> f = Facility.objects.get(pk='CN2019178FQ1RKZ')
>>> f.c
f.check(           f.clean(           f.clean_fields(    f.contributors(    f.country_code     f.created_at       f.created_from     f.created_from_id  
>>> f.created_from
<FacilityListItem: FacilityListItem 594 - MATCHED>
>>> f.created_from.processing_results
[{'error': False, 'action': 'parse', 'started_at': '2019-06-26 18:54:46.654964', 'finished_at': '2019-06-26 18:54:46.659191', 'is_geocoded': True}, {'error': False, 'action': 'geocode', 'started_at': '2019-06-26 18:54:51.556838', 'finished_at': '2019-06-26 18:54:51.556877', 'skipped_geocoder': True}, {'error': False, 'action': 'match', 'started_at': '2019-06-26 18:54:52.275275', 'finished_at': '2019-06-26 18:55:10.561837'}, {'error': False, 'action': 'merge_facility', 'started_at': '2019-06-26 18:59:01.103755', 'finished_at': '2019-06-26 18:59:01.103755', 'merged_oar_id': 'CN2019177A7GXDQ'}, {'error': False, 'action': 'merge_facility', 'started_at': '2019-06-26 19:25:29.522594', 'finished_at': '2019-06-26 19:25:29.522594', 'merged_oar_id': 'CN2019177ZT0K3V'}, {'error': False, 'action': 'split_facility', 'started_at': '2019-06-27 01:08:15.123850', 'finished_at': '2019-06-27 01:08:15.123850', 'previous_facility_oar_id': 'CN2019177D9KY2N'}]
```

**Comment by kellyi** [comment]
Thanks!

**Pull request merged**